### PR TITLE
DOC: add interp example

### PR DESCRIPTION
**Pull request by raybellwaves**
- [NA] Closes #xxxx (remove if there is no corresponding issue, which should only be the case for minor changes)
 - [NA] Tests added (for all bug fixes or enhancements)
 - [NA] Tests passed (for all non-documentation changes)
 - [NA] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API (remove if this change should not be visible to users, e.g., if it is an internal clean-up, or if this is part of a larger project that will be documented later)

I wanted to add an example as googling 'xarray interp' takes me to http://xarray.pydata.org/en/stable/generated/xarray.Dataset.interp.html.

However, there lots of great examples in http://xarray.pydata.org/en/stable/interpolation.html
Perhaps I should edit it to:
```
Examples
--------
See http://xarray.pydata.org/en/stable/interpolation.html
```

I was also thinking about adding a note about interpolating time as it came up in https://github.com/pydata/xarray/issues/2284 but not sure what would be the best way to do so.

### CODE REVIEW

**Comment by fujiisoup** [comment]
Thanks for adding this to doc.
It is very welcome if you could add an example for interp with datetime.

To do so, it would be best place to put it in here
https://github.com/pydata/xarray/blob/master/doc/interpolation.rst

**Comment by raybellwaves** [comment]
@fujiisoup I would like to add another example on how to update a `DataArray` with the interpolated data. I looked in http://xarray.pydata.org/en/stable/combining.html but couldn't work out how best to do it.

I tried using `merge` but it requires the `DataArray`'s to be named.

I used `concat` in this example but it is not correct as I would like the coordinates to be in order (in-place may be the right word). 

```
import xarray as xr
import pandas as pd
import numpy as np

da_dt64 = xr.DataArray([1, 3], [('time', pd.date_range('1/1/2000', '1/3/2000', periods=2))])
a = da_dt64.interp(time=np.datetime64('2000-01-02'))
xr.concat([da_dt64, a], dim='time')
#<xarray.DataArray (time: 3)>
#array([ 1.,  3.,  2.])
#Coordinates:
#  * time     (time) datetime64[ns] 2000-01-01 2000-01-03 2000-01-02
```

**Comment by fujiisoup** [comment]
Thanks.

I think you can do
```python
da_dt64.interp(time=pd.date_range('1/1/2000', '1/3/2000', periods=3))
```

File: doc/interpolation.rst [hunk 1]
```diff
@@ -48,6 +48,23 @@ array-like, which gives the interpolated result as an array.
     # interpolation
     da.interp(time=[2.5, 3.5])
 
+
+To interpolate data with a :py:func:`numpy.datetime64` coordinate you have to 
+wrap it explicitly in a :py:func:`numpy.datetime64` object.
```
Comment (fujiisoup) [inline doc/interpolation.rst]: It is currently true, but I think we could support `da_dt64.interp(time='2000-01-02')` in the future.
Do you mind to send another PR to do this?

Comment (raybellwaves) [reply doc/interpolation.rst]: I'm trying to wrap up another PR. Unfortunately, I don't want to over promise and may not be able to get to this until a couple of weeks. If you have ideas of how to do this we can move the discussion to https://github.com/pydata/xarray/issues/2284

**Comment by fujiisoup** [comment]
Thanks, @raybellwaves.

**Pull request merged**